### PR TITLE
Weather API interface

### PR DESCRIPTION
**Pull request by brennanholtzclaw**
Users should be able to enter their location and be given a forecast. 
If it is predicted to rain, a special RAINBOW offer will be available!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ gem 'pg'
 gem "paperclip", "~> 4.2"
 gem 'rails_12factor', group: :production
 group :development, :test do
+  gem 'faraday'
   gem 'pry'
```
Comment (erinnachen) [inline Gemfile]: Should this be in production env as well?

File: app/assets/javascripts/weather.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                           1000 );
+  });
+  $( ".pot-o-gold" ).on("mouseleave", function() {
+    $( ".grow" ).animate({ width: "100%",
+                           height: "200px",
+                           left: "+=50" },
+                           1000 );
+    $(this).toggleClass("grow");
+  });
+
```
Comment (erinnachen) [inline app/assets/javascripts/weather.js]: We could think about refactoring this into two methods?

File: app/controllers/weather_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  attr_accessor :city
+  def show
+    weather_key = ENV["OPEN_WEATHER_KEY"]
+    @city = session[:city]
+    result = Faraday.get("http://api.openweathermap.org/data/2.5/forecast?q=#{session[:city].downcase},us&units=imperial&APPID=#{weather_key}&mode=json")
+    @forecast = JSON.parse(result.body, object_class: OpenStruct)
+    @forecast = @forecast.list.select do |day_part|
+      DateTime.parse(day_part.dt_txt).strftime("%k %p") == "15 PM"
+    end[0..3]
+    # binding.pry
```
Comment (erinnachen) [inline app/controllers/weather_controller.rb]: Should remove this

File: app/controllers/weather_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def new
+  end
+
+  def create
+    session[:city] = params[:weather][:us_city]
+    redirect_to weather_path(1)
```
Comment (erinnachen) [inline app/controllers/weather_controller.rb]: will this weather_path always have id 1?

File: app/views/shared/_categories_nav.html.erb [hunk 5]
```diff
@@ -12,6 +12,7 @@
       <li>
         <%= link_to "all items", items_path, class: "green-font" %>
       </li>
+      <li class="mousiness"><a href="<%= new_weather_path %>" class="green-font">Check the Weather</a></li>
```
Comment (erinnachen) [inline app/views/shared/_categories_nav.html.erb]: mousiness is no longer need.

File: app/views/weather/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div class="well text-center">
+  <%= form_for(:weather, url: weather_index_path) do |f| %>
+    <h4><%= f.label :us_city, "US City" %></h4>
+    <h4><%= f.text_field :us_city %></h4>
+    <%= f.submit "Check Weather", class: "btn btn-success mousiness btn-font-size" %>
+  <% end %>
```
Comment (erinnachen) [inline app/views/weather/new.html.erb]: mousiness no longer needed

File: config/routes.rb [hunk 7]
```diff
@@ -20,6 +20,9 @@
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy"
 
+  resources :weather, only: [:create, :show, :new]
+  # resources :weather, only: [:new, :show, :create]
+
```
Comment (erinnachen) [inline config/routes.rb]: remove this comment

File: app/views/weather/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+      <a href="/items/<%= Item.find_by(title: "Pot O' Gold").id %>"><%= image_tag "pot_o_gold.png", class: "pot-o-gold" %></a>
+    <% elsif day_part.weather.first.description.include?("clear sky")%></a>
+      <a href="/items/<%= Item.find_by(title: "Four-Leaf Clover").id %>"><%= image_tag "sunshine.png", class: "sunshine" %></a>
+    <% else %>
+      <a href="/items/<%= Item.find_by(title: "Uncatchable Leprechaun").id %>"><%= image_tag "leprechaun_face.png", class: "leprechaun" %></a>
+    <% end %>
+  </div>
+<% end %>
+</div>
+
```
Comment (erinnachen) [inline app/views/weather/show.html.erb]: There seems to be a fair bit of logic and ActiveRecord calls in here. Probably an area to target for refactoring

File: app/controllers/weather_controller.rb (hunk 4 above)
Comment (brennanholtzclaw) [inline app/controllers/weather_controller.rb]: Yes it will.

File: app/views/weather/show.html.erb (hunk 8 above)
Comment (brennanholtzclaw) [inline app/views/weather/show.html.erb]: Pulled those calls out into the Item model.

**Comment by erinnachen** [comment]
This is an awesome feature!

**Pull request merged**